core/src/test/java/io/fabric8/maven/core/handler/JobHandlerTest.java

```diff
@@ -0,0 +1,172 @@
+        volumes1.add(volumeConfig1);
+
+        //container name with alias
+        BuildImageConfiguration buildImageConfiguration = new BuildImageConfiguration.Builder().
+                ports(ports).from("fabric8/maven:latest").cleanup("try")
+                .tags(tags).compression("gzip").build();
+
+        ImageConfiguration imageConfiguration = new ImageConfiguration.Builder().
+                name("test").alias("test-app").buildConfig(buildImageConfiguration)
+                .registry("docker.io").build();
+
+        images.add(imageConfiguration);
+    }
+
+    @Test
+    public void jobHandlerTest() {
+
+        ContainerHandler containerHandler =
+                new ContainerHandler(project, envVarHandler, probeHandler);
+
+        PodTemplateHandler podTemplateHandler = new PodTemplateHandler(containerHandler);
+
+        JobHandler jobHandler = new JobHandler(podTemplateHandler);
+
+        ResourceConfig config = new ResourceConfig.Builder()
+                .imagePullPolicy("IfNotPresent")
+                .controllerName("testing")
+                .withServiceAccount("test-account")
+                .volumes(volumes1)
+                .build();
+
+        Job job = jobHandler.getJob(config,images);
+
+        //Assertion
+        assertNotNull(job.getSpec());
+        assertNotNull(job.getMetadata());
+        assertNotNull(job.getSpec().getTemplate());
+        assertEquals("testing",job.getMetadata().getName());
+        assertEquals("test-account",job.getSpec().getTemplate()
+                .getSpec().getServiceAccountName());
+        assertFalse(job.getSpec().getTemplate().getSpec().getVolumes().isEmpty());
+        assertEquals("test",job.getSpec().getTemplate().getSpec().
+                getVolumes().get(0).getName());
+        assertEquals("/test/path",job.getSpec().getTemplate()
+                .getSpec().getVolumes().get(0).getHostPath().getPath());
+        assertNotNull(job.getSpec().getTemplate().getSpec().getContainers());
+
+    }
+
+    @Test
+    //invalid controller name
+    public void daemonTemplateHandlerSecondTest() {
+        try {
+            ContainerHandler containerHandler =
+                    new ContainerHandler(project, envVarHandler, probeHandler);
+
+            PodTemplateHandler podTemplateHandler = new PodTemplateHandler(containerHandler);
+
+            JobHandler jobHandler = new JobHandler(podTemplateHandler);
+
+            //with invalid controller name
+            ResourceConfig config = new ResourceConfig.Builder()
+                    .imagePullPolicy("IfNotPresent")
+                    .controllerName("TesTing")
+                    .withServiceAccount("test-account")
+                    .volumes(volumes1)
+                    .build();
+
+            jobHandler.getJob(config, images);
+        }
+        //asserting the exact message because
+        // it throws the same exception in case controller name is null
+        catch(IllegalArgumentException exception) {
```

Comment:
Instead of try-catch use "@Test(expected=IllegalArgumentException.class) annotation"